site/src/xServices/auth/authXService.ts

```diff
@@ -469,24 +511,27 @@ export const authMachine =
         signIn: async (_, event) => {
           return await API.login(event.email, event.password)
         },
-        signOut: async () => {
+        checkHost: async () => {
+          const appHost = await API.getApplicationsHost()
+          return appHost.host
+        },
+        signOutAndRedirect: async (context) => {
           // Get app hostname so we can see if we need to log out of app URLs.
           // We need to load this before we log out of the API as this is an
           // authenticated endpoint.
-          const appHost = await API.getApplicationsHost()
           await API.logout()
-
-          if (appHost.host) {
+          if (context.host) {
```

Comment:
For some reason, host isn't on context. Any ideas @BrunoQuaresma or @presleyp? I assign it on ln 642.

Reply:
No idea but I think it was because we thought it was only used by a page and not for the application overall? 🤔

Reply:
If you think we should add it there, go for it!